plotAfb.py

```diff
@@ -232,7 +240,93 @@ def plotDataMC(plotsPos,plotsNeg, background):
+								histTree[flavor][year][dataType][category].append(tempHistAfb.Clone())
+                                                elif flavor == "electron" and category == "BB":
+							histTree[flavor][year][dataType][category]=[]
+							for process in signalTree[flavor][year]:
+								tempHistPos=process.loadHistogram(plotTree[flavor][category][0],lumi[flavor][year],zScaleFac[flavor][year][1])
+								tempHistNeg=process.loadHistogram(plotTree[flavor][category][1],lumi[flavor][year],zScaleFac[flavor][year][1])
+								tempNom=tempHistPos.Clone()
+								tempDenom=tempHistPos.Clone()
+								tempNom.Add(tempHistNeg,-1)
+								tempDenom.Add(tempHistNeg)
+								tempDenom.Sumw2()
+								tempNom.Sumw2()
+								tempNom=tempNom.Rebin(len(bng) - 1, 'hist_' + uuid.uuid4().hex, array('d', bng))
+								tempDenom=tempDenom.Rebin(len(bng) - 1, 'hist_' + uuid.uuid4().hex, array('d', bng))
+								tempHistAfb=ROOT.TH1D("Afb"+flavor+year+category+dataType,"The distribution of Afb for "+year+" "+flavor+" result",len(binning(flavor))-1  ,array("f",binning(flavor)))
+								tempHistAfb.Divide(tempNom,tempDenom)
+								histTree[flavor][year][dataType][category].append(tempHistAfb.Clone())
+						else:
+							histTree[flavor][year][dataType][category]=[]
+                                                        for process in signalTree[flavor][year]:
+                                                                tempHistPos=process.loadHistogram(plotTree[flavor][category][0],lumi[flavor][year],zScaleFac[flavor][year])
+                                                                tempHistNeg=process.loadHistogram(plotTree[flavor][category][1],lumi[flavor][year],zScaleFac[flavor][year])
+                                                                tempNom=tempHistPos.Clone()
+								tempNom.Add(tempHistNeg,-1)
+                                                                tempDenom=tempHistPos.Clone()
+                                                                tempDenom.Add(tempHistNeg)
+                                                                tempDenom.Sumw2()
+                                                                tempNom.Sumw2()
+                                                                tempNom=tempNom.Rebin(len(bng) - 1, 'hist_' + uuid.uuid4().hex, array('d', bng))
+                                                                tempDenom=tempDenom.Rebin(len(bng) - 1, 'hist_' + uuid.uuid4().hex, array('d', bng))
+                                                                tempHistAfb=ROOT.TH1D("Afb"+flavor+year+category+dataType,"The distribution of Afb for "+year+" "+flavor+" result",len(binning(flavor))-1  ,array("f",binning(flavor)))
+                                                                tempHistAfb.Divide(tempNom,tempDenom)
+                                                                histTree[flavor][year][dataType][category].append(tempHistAfb.Clone())			
```

Comment:
This looks like you are not subtracting the 100k TeV samples, right?

Reply:
I used 100 kTeV sample.